### PR TITLE
borer 1.1

### PR DESCRIPTION
**Pull request by jjpark-kb**
## About The Pull Request
Warning: This is a lot.

1) Borers have universal objectives again
1a) Five borers need to at least produce 10 eggs
1b) Twenty hosts need to agree to be a willing host
1c) Completing objectives will boost your evolution point gain
2) Some borer abilities have been locked away and require unlocking
2a) Revive host, learn chemical through blood, willing host, and produce offspring
3) Borers will screw with your health hud, similar to miner's salve
4) Borers can only have 10 evolution points (so 10 chem and 10 stat) at a time
4a) It will tell you that you were unable to gain a point. Spend points to earn more
5) Hosts will naturally take toxin damage up to 25
6) There is a distinction between the link and the hivemind
6a) The hivemind is purple as well, but is now bold
7) Some abilities got new icons
8) Checking the "blood" (read: health) of a host will now display like a health analyzer
9) Borers can enter dead bodies
10) Borers no longer spawn other borers, but instead eggs
10a) They have some information for the beginner borer on spawners menu

Objectives:
image
image

Some new ability icons:
image

Unable to gain over the limit:
image

Egg:
image

Link vs. Hivemind
image

## How This Contributes To The Skyrat Roleplay Experience
This changes borers a lot. There are some buffs and some nerfs.
People argued (and it was valid) that there was not enough drawbacks to allowing the borer inside of you; this should solve some of those arguments for now.
Borers should be limited in some ways and I hope that what I have done can make it feel more rewarding to play borer.

## Changelog

:cl:
add: borers have universal objectives again
add: borers now give the screwy health hud effect (similar to miner's salve)
add: borer hosts will naturally take toxin damage, up to 25
add: borers will now spawn eggs rather than complete borers
balance: some borer abilities have been locked behind stat evolution points
balance: borers can only have 10 chemical and 10 stat evolution points
balance: borers when checking health of host get a similar output as a health analyzer
balance: borers can now enter dead bodies
qol: a distinction between link and hivemind for borers: hivemind is bold now
imageadd: some borer abilities got new icons
/:cl:

### CODE REVIEW

File: modular_skyrat/modules/cortical_borer/code/cortical_borer.dm [hunk 1]
```diff
@@ -1,3 +1,6 @@
+GLOBAL_VAR_INIT(borer_spawner_ten, 0)
```
Comment (YakumoChen) [inline modular_skyrat/modules/cortical_borer/code/cortical_borer.dm]: Don't hardcode the amount

File: modular_skyrat/modules/cortical_borer/code/cortical_borer.dm [hunk 2]
```diff
@@ -302,38 +312,67 @@
 	. += "Chemical Storage: [chemical_storage]/[max_chemical_storage]"
 	. += "Chemical Evolution Points: [chemical_evolution]"
 	. += "Stat Evolution Points: [stat_evolution]"
+	. += ""
 	if(host_sugar())
 		. += "Sugar detected! Unable to generate resources!"
+		. += ""
+	. += "OBJECTIVES:"
+	. += "1) Five(5) borers producing ten(10) eggs: [GLOB.borer_spawner_ten]/5"
```
Comment (YakumoChen) [inline modular_skyrat/modules/cortical_borer/code/cortical_borer.dm]: These should be just numerals, and they should use a define for the amount.

File: modular_skyrat/modules/cortical_borer/code/cortical_borer.dm [hunk 3]
```diff
@@ -302,38 +312,67 @@
...
+	. += "2) Twenty(20) willing hosts: [length(GLOB.willing_hosts)]/20"
```
Comment (YakumoChen) [inline modular_skyrat/modules/cortical_borer/code/cortical_borer.dm]: Ditto

File: modular_skyrat/modules/cortical_borer/code/cortical_borer_egg.dm [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+	name = "borer egg"
+	desc = "An egg of a creature that is known to crawl inside of you, be careful."
+	icon = 'modular_skyrat/modules/cortical_borer/icons/animal.dmi'
+	icon_state = "brainegg"
+	layer = BELOW_MOB_LAYER
+	density = FALSE
+	mob_name = "cortical borer"
+	///Type of mob that will be spawned
+	mob_type = /mob/living/simple_animal/cortical_borer
+	role_ban = ROLE_DRONE
```
Comment (YakumoChen) [inline modular_skyrat/modules/cortical_borer/code/cortical_borer_egg.dm]: Borers are an antag, they should be using a proper antagonist ban type (like xeno)

**Review by YakumoChen — CHANGES_REQUESTED** [review]

**Review by Gandalf2k15 — APPROVED** [review]

**Pull request merged**